plugin/pkg/admission/network/defaultingressclass/admission_test.go

```diff
@@ -196,3 +196,242 @@ func TestAdmission(t *testing.T) {
 		})
 	}
 }
+
+func TestAdmissionUpdated(t *testing.T) {
+	defaultClass1 := &networkingv1.IngressClass{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "IngressClass",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default1",
+			Annotations: map[string]string{
+				networkingv1.AnnotationIsDefaultIngressClass: "true",
+			},
+		},
+	}
+	defaultClass2 := &networkingv1.IngressClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default2",
+			Annotations: map[string]string{
+				networkingv1.AnnotationIsDefaultIngressClass: "true",
+			},
+		},
+	}
+	// Class that has explicit default = false
+	classWithFalseDefault := &networkingv1.IngressClass{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "IngressClass",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "nondefault1",
+			Annotations: map[string]string{
+				networkingv1.AnnotationIsDefaultIngressClass: "false",
+			},
+		},
+	}
+	// Class with missing default annotation (=non-default)
+	classWithNoDefault := &networkingv1.IngressClass{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "IngressClass",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "nondefault2",
+		},
+	}
+	// Class with empty default annotation (=non-default)
+	classWithEmptyDefault := &networkingv1.IngressClass{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "IngressClass",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "nondefault2",
+			Annotations: map[string]string{
+				networkingv1.AnnotationIsDefaultIngressClass: "",
+			},
+		},
+	}
+
+	testCases := []struct {
+		name                  string
+		classesBefore         []*networkingv1.IngressClass
```

Comment:
From what I can tell this isn't actually used? Maybe just create a single "classes" field instead of "classesAfter"?